### PR TITLE
various documentation edits

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [N/A] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [N/A] Have you successfully run `brew style` with your changes locally?
- [N/A] Have you successfully run `brew tests` with your changes locally?

-----
- fixes for formatting, grammar, wording for consistency & readability
- fix styling, flag order of sample commands
- monospace text of links to files
- check & update GitHub file links
- de-Oxfordise some spellings
- move & edit `.editorconfig` to apply to Markdown files

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -203,7 +203,7 @@ our analytics identified it was not widely used.
 @mxcl was too concerned with the beer theme and didn’t consider that the
 project may actually prove popular. By the time he realised it was, it was too
 late. However, today, the first Google hit for “homebrew” is not beer
-related ;-)
+related ;&#8209;)
```
Comment (EricFromCanada) [inline docs/FAQ.md]: A non-breaking hyphen here keeps the emoticon together when output to HTML.

File: docs/Formula-Cookbook.md [hunk 2]
```diff
@@ -269,20 +269,20 @@ Some software installs to `man` instead of `share/man`, so check the output and
...
 
-If you’re not sure about the name check the homepage, and check the Wikipedia page and [what Debian calls it](https://www.debian.org/distrib/packages).
+If you’re not sure about the name, check its homepage, Wikipedia page and [what Debian calls it](https://www.debian.org/distrib/packages).
 
 When Homebrew already has a formula called `foo` we typically do not accept requests to replace that formula with something else also named `foo`. This is to avoid both confusing and surprising users’ expectations.
 
-When two formulae share an upstream name, e.g. [`AESCrypt`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aescrypt.rb) and [`AESCrypt`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aescrypt-packetizer.rb) the newer formula must typically adapt the name to avoid conflict with the current formula.
+When two formulae share an upstream name, e.g. [AESCrypt](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aescrypt.rb) and [AES Crypt](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aescrypt-packetizer.rb) the newer formula must typically adapt its name to avoid conflict with the current formula.
```
Comment (EricFromCanada) [inline docs/Formula-Cookbook.md]: In this case the link text for both is the upstream package's name to highlight their similarity.

File: docs/Acceptable-Formulae.md [hunk 3]
```diff
@@ -15,15 +15,15 @@ We now accept versioned formulae as long as they [meet the requirements](Version
...
+### We don’t like install scripts that download unversioned things
 We don't like install scripts that are pulling from the `master` branch of Git repositories or unversioned, unchecksummed tarballs. These should use `resource` blocks with specific revisions or checksummed tarballs instead. Note that we now allow tools like `cargo`, `gem` and `pip` to download specifically versioned libraries during installation.
 
 ### We don’t like binary formulae
 Our policy is that formulae in the core tap
 ([homebrew/core](https://github.com/Homebrew/homebrew-core)) must be open-source
 and either built from source or produce cross-platform binaries (e.g. Java, Mono).
 Binary-only formulae should go to
-[Homebrew Cask](https://github.com/Homebrew/homebrew-cask).
+[homebrew/cask](https://github.com/Homebrew/homebrew-cask).
```
Comment (colindean) [inline docs/Acceptable-Formulae.md]: I think `Homebrew Cask` is the proper name…

File: docs/Acceptable-Formulae.md [hunk 4]
```diff
@@ -57,16 +57,16 @@ running a package manager.
 
 ### Stuff that builds an `.app`
 Don’t make your formula build an `.app` (native macOS Application); we
-don’t want those things in Homebrew. Encourage upstream projects to build and support a `.app` that can be distributed by [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) (and used without it, too).
+don’t want those things in Homebrew. Encourage upstream projects to build and support a `.app` that can be distributed by [homebrew/cask](https://github.com/Homebrew/homebrew-cask) (and used without it, too).
```
Comment (colindean) [inline docs/Acceptable-Formulae.md]: I think `Homebrew Cask` is the proper name…

File: docs/Acceptable-Formulae.md [hunk 5]
```diff
@@ -57,16 +57,16 @@ running a package manager.
...
 
 ### Stuff that builds a GUI by default (but doesn't have to)
-Make it build a command-line tool or a library by default and, if the GUI is useful and would be widely used, add an option to build the GUI. Don't offer an option for multiple GUI backends e.g. X11 is a bad user experience for GUIs on macOS.
+Make it build a command-line tool or a library by default and, if the GUI is useful and would be widely used, add an option to build the GUI. Don't offer an option for multiple GUI backends, e.g. X11 is a bad user experience for GUIs on macOS.
```
Comment (colindean) [inline docs/Acceptable-Formulae.md]: I was going to recommend another comma after X11, but then I see that this sentence can probably be reworded:

> Don't offer an option for multiple GUI backends. Avoid X11, as it is a bad user experience for GUIs on macOS.

File: docs/Formula-Cookbook.md [hunk 6]
```diff
@@ -245,7 +245,7 @@ brew install --verbose --debug foo
 
 `--debug` will ask you to open an interactive shell if the build fails so you can try to figure out what went wrong.
 
-Check the top of the e.g. `./configure` output. Some configure scripts do not recognize e.g. `--disable-debug`. If you see a warning about it, remove the option from the formula.
+Check the top of the e.g. `./configure` output. Some configure scripts do not recognise e.g. `--disable-debug`. If you see a warning about it, remove the option from the formula.
```
Comment (colindean) [inline docs/Formula-Cookbook.md]: 🇺🇸 et al. vs 🇬🇧 et al. What does the project prefer?

File: docs/Formula-Cookbook.md [hunk 7]
```diff
@@ -691,7 +691,7 @@ The symlinks created by [`install_symlink`](https://www.rubydoc.info/github/Home
 
 ### Handling files that should persist over formula upgrades
 
-For example, Ruby 1.9’s gems should be installed to `var/lib/ruby/` so that gems don’t need to be reinstalled when upgrading Ruby. You can usually do this with symlink trickery, or *better* a configure option.
+For example, Ruby 1.9’s gems should be installed to `var/lib/ruby/` so that gems don’t need to be reinstalled when upgrading Ruby. You can usually do this with symlink trickery, or (ideally) a configure option.
```
Comment (colindean) [inline docs/Formula-Cookbook.md]: You can usually do this with symlink trickery, or (ideally) a configure option.

could be

    You can usually do this with a configure option, or symlink trickery, alternatively.

Placing the preferred one first and then saying the other is an alternative will lead people to the first in most cases.

File: docs/Python-for-Formula-Authors.md [hunk 8]
```diff
@@ -143,29 +143,29 @@ If the configure script takes a `--with-python` flag, it usually will not need e
...
+Sometimes we have to edit a `Makefile` on-the-fly to use our prefix for the Python bindings using Homebrew's [`inreplace`](Formula-Cookbook.md#inreplace) helper method.
 
 ## Libraries
 
 ### Python declarations
 
 Libraries built for Python 3 should include `depends_on "python"`, which will bottle against Homebrew's Python 3.x. Python 2.x libraries must function when they are installed against either the system Python or brewed Python.
 
-Python 2 libraries do not need a `depends_on "python@2"` declaration; they will be built with the system Python, but should still be usable with any other Python 2.7. If this is not the case, it is an upstream bug; [here is some advice for resolving it](https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/).
+Python 2 libraries do not need a `depends_on "python@2"` declaration; they will be built with the system Python, but should still be usable with any other Python 2.7. If this is not the case, it's an upstream bug; [here's some advice for resolving it](https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/).
```
Comment (colindean) [inline docs/Python-for-Formula-Authors.md]: I don't really like contractions in tech writing but if there's nothing else to change in this file, leave it.

File: docs/Formula-Cookbook.md (hunk 6 above)
Comment (EricFromCanada) [reply docs/Formula-Cookbook.md]: The [style guide](https://docs.brew.sh/Prose-Style-Guidelines) prefers Commonwealth English, as do [previous commits](https://github.com/Homebrew/brew/commit/c1376f9f18f6e5da3d2290d5cdd808d32767068b#diff-e3e2a9bfd88566b05001b02a3f51d286).

File: docs/Python-for-Formula-Authors.md (hunk 8 above)
Comment (EricFromCanada) [reply docs/Python-for-Formula-Authors.md]: This matches the wording in the Bindings section earlier in the document.

File: docs/Acceptable-Formulae.md (hunk 3 above)
Comment (MikeMcQuaid) [reply docs/Acceptable-Formulae.md]: @EricFromCanada Would you be interested in working on that integration at all?

**Comment by MikeMcQuaid** [comment]
Thanks again @EricFromCanada, you keep on rocking!

**Pull request merged**